discovery_engine_core/web-api/src/models.rs

```diff
@@ -91,23 +113,19 @@ pub(crate) struct InteractionRequestBody {
 pub(crate) struct UserId(String);
 
 impl UserId {
-    fn new(value: &str) -> Result<Self, Error> {
-        let value = urlencoding::decode(value).map_err(Error::UserIdUtf8Conversion)?;
-
-        if value.trim().is_empty() {
-            Err(Error::UserIdEmpty)
-        } else if value.contains('\u{0000}') {
-            Err(Error::UserIdContainsNul)
-        } else {
-            Ok(Self(value.to_string()))
-        }
+    fn new(id: String) -> Result<Self, IdValidationError> {
+        let id = urlencoding::decode(&id).map_err(IdValidationError::InvalidUtf8)?;
```

Comment:
No actually we need to remove it from the `UserId` as well since we are requiring that a `UserId` can only be created from a urlencoded string which is not the case.